Errors when passing `ArrayBuffer` into `write`/`writeAsync`

```
k6 v1.7.1 (go1.26.1, linux/amd64)
Extensions:
  github.com/grafana/xk6-tcp v0.1.0, k6/x/tcp [js]
```

---

Error 1: `write`:
```javascript
import { Socket } from "k6/x/tcp";

export default function() {
  const socket = new Socket();
  socket.write(new ArrayBuffer(4));
}
```
Results in a segfault:
<details>

```
> k6 run repro.js

         /\      Grafana   /‾‾/  
    /\  /  \     |\  __   /  /   
   /  \/    \    | |/ /  /   ‾‾\ 
  /          \   |   (  |  (‾)  |
 / __________ \  |_|\_\  \_____/ 


     execution: local
        script: repro.js
        output: -

     scenarios: (100.00%) 1 scenario, 1 max VUs, 10m30s max duration (incl. graceful stop):
              * default: 1 iterations for each of 1 VUs (maxDuration: 10m0s, gracefulStop: 30s)

panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xf4044a]

goroutine 42 [running]:
github.com/grafana/sobek.(*Runtime).runWrapped.func1()
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/runtime.go:2525 +0xd9
panic({0x1bb3fe0?, 0x33ac160?})
        runtime/panic.go:860 +0x13a
github.com/grafana/sobek.(*vm).handleThrow(0x27f6ee454000, {0x1bb3fe0, 0x33ac160})
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/vm.go:847 +0x425
github.com/grafana/sobek.(*vm).try.func1()
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/vm.go:866 +0x3a
panic({0x1bb3fe0?, 0x33ac160?})
        runtime/panic.go:860 +0x13a
github.com/grafana/sobek.(*vm).handleThrow(0x27f6ee454000, {0x1bb3fe0, 0x33ac160})
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/vm.go:847 +0x425
github.com/grafana/sobek.(*vm).runTryInner.func1()
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/vm.go:889 +0x3a
panic({0x1bb3fe0?, 0x33ac160?})
        runtime/panic.go:860 +0x13a
github.com/grafana/xk6-tcp/tcp.(*socket).write(0x27f6ee1aa608, {0x2037a78?, 0x27f6ee067800?}, 0x27f6ee1e6090?)
        github.com/grafana/xk6-tcp@v0.1.0/tcp/socket_write.go:26 +0x4a
reflect.Value.call({0x1b65240?, 0x27f6ee3a6bb0?, 0x27f6ee40d200?}, {0x1ea996f, 0x4}, {0x27f6ee067860, 0x2, 0x27f6ee40d228?})
        reflect/value.go:586 +0xf06
reflect.Value.Call({0x1b65240?, 0x27f6ee3a6bb0?, 0x27f6ee067800?}, {0x27f6ee067860?, 0x27f6ee3a6f40?, 0x7fe9f3d638e8?})
        reflect/value.go:369 +0xb9
github.com/grafana/sobek.(*Runtime).newWrappedFunc.(*Runtime).wrapReflectFunc.func1({{0x2037a78, 0x27f6ee1e33b0}, {0x27f6ee0d5320, 0x1, 0x6}})
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/runtime.go:2044 +0x405
github.com/grafana/sobek.(*nativeFuncObject).vmCall(0x27f6ee277d40, 0x27f6ee454000, 0x1)
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/func.go:571 +0x184
github.com/grafana/sobek.call.exec(0x2061da0?, 0x27f6ee454000)
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/vm.go:3732 +0x66
github.com/grafana/sobek.(*vm).run(0x27f6ee454000)
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/vm.go:642 +0x5b
github.com/grafana/sobek.(*vm).runTryInner(0x27f6ee454000?)
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/vm.go:893 +0x52
github.com/grafana/sobek.(*baseJsFuncObject).__call(0x27f6ee277800, {0x27f6ee3a6ae0?, 0x1, 0x27f6ee450488?}, {0x0, 0x0}, {0x2038128?, 0x341a7e0?})
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/func.go:434 +0x705
github.com/grafana/sobek.(*baseJsFuncObject)._call(...)
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/func.go:450
github.com/grafana/sobek.(*baseJsFuncObject).call(0x27f6ee05f6e0?, {{0x2038128, 0x341a7e0}, {0x27f6ee3a6ae0, 0x1, 0x1}}, {0x0?, 0x0?})
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/func.go:458 +0x6d
github.com/grafana/sobek.(*baseJsFuncObject).Call(...)
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/func.go:390
github.com/grafana/sobek.AssertFunction.func1.1()
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/runtime.go:2485 +0x56
github.com/grafana/sobek.(*vm).try(0x27f6ee454000, 0x27f6ee05f8e8)
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/vm.go:870 +0x21b
github.com/grafana/sobek.(*Runtime).runWrapped(0x27f6ee450008, 0x48aea5?)
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/runtime.go:2529 +0x5f
github.com/grafana/sobek.AssertFunction.func1({0x2038128?, 0x341a7e0?}, {0x27f6ee3a6ae0?, 0x1?, 0x1?})
        github.com/grafana/sobek@v0.0.0-20260219184149-bdae4a158e94/runtime.go:2484 +0x74
go.k6.io/k6/internal/js.(*VU).runFn.func2()
        go.k6.io/k6@v1.7.1/internal/js/runner.go:939 +0x3e
go.k6.io/k6/internal/js/eventloop.(*EventLoop).Start(0x27f6ee35a000, 0x27f6ee1e31d0)
        go.k6.io/k6@v1.7.1/internal/js/eventloop/eventloop.go:179 +0x133
go.k6.io/k6/internal/js.(*VU).runFn(0x27f6ee436e60, {0x201b150, 0x27f6ee1fc1e0}, 0x1, 0x27f6ee3284e0, 0x27f6ee3a6ac0, {0x27f6ee3a6ae0, 0x1, 0x1})
        go.k6.io/k6@v1.7.1/internal/js/runner.go:938 +0x270
go.k6.io/k6/internal/js.(*ActiveVU).RunOnce(0x27f6ee456440)
        go.k6.io/k6@v1.7.1/internal/js/runner.go:871 +0x4c5
go.k6.io/k6/lib/executor.PerVUIterations.Run.getIterationRunner.func7({0x201b118, 0x27f6ee1e2f90}, {0x200a860?, 0x27f6ee456440?})
        go.k6.io/k6@v1.7.1/lib/executor/helpers.go:81 +0x4a
go.k6.io/k6/lib/executor.PerVUIterations.Run.func5({0x2012348, 0x27f6ee436e60})
        go.k6.io/k6@v1.7.1/lib/executor/per_vu_iterations.go:228 +0x434
created by go.k6.io/k6/lib/executor.PerVUIterations.Run in goroutine 40
        go.k6.io/k6@v1.7.1/lib/executor/per_vu_iterations.go:241 +0xce5
```

</details>

---

Error 2: `writeAsync`:
```javascript
import { Socket } from "k6/x/tcp";

export default async function() {
  const socket = new Socket();
  await socket.writeAsync(new ArrayBuffer(4));
}
```
Throws an error `invalid type: String or ArrayBuffer expected`, even though it was passed an `ArrayBuffer`.

<details>

```
> k6 run repro.js

         /\      Grafana   /‾‾/  
    /\  /  \     |\  __   /  /   
   /  \/    \    | |/ /  /   ‾‾\ 
  /          \   |   (  |  (‾)  |
 / __________ \  |_|\_\  \_____/ 


     execution: local
        script: repro.js
        output: -

     scenarios: (100.00%) 1 scenario, 1 max VUs, 10m30s max duration (incl. graceful stop):
              * default: 1 iterations for each of 1 VUs (maxDuration: 10m0s, gracefulStop: 30s)

ERRO[0000] Uncaught (in promise) GoError: invalid type: String or ArrayBuffer expected
        at github.com/grafana/xk6-tcp/tcp.(*socket).writeAsync-fm (native)
        at default (file:///.../repro.js:5:26(9))  executor=per-vu-iterations scenario=default


  █ TOTAL RESULTS 

    EXECUTION
    iteration_duration...: avg=136.71µs min=136.71µs med=136.71µs max=136.71µs p(90)=136.71µs p(95)=136.71µs
    iterations...........: 1   4233.700254/s

    NETWORK
    data_received........: 0 B 0 B/s
    data_sent............: 0 B 0 B/s




running (00m00.0s), 0/1 VUs, 1 complete and 0 interrupted iterations
default ✓ [======================================] 1 VUs  00m00.0s/10m0s  1/1 iters, 1 per VU
```

</details>

---

Both reproductions run with expected errors when passed a string instead of an `ArrayBuffer` ("no active connection"), so I am sure that those are `ArrayBuffer`-related issues.

--- a/examples/binary.js
+++ b/examples/binary.js
@@ -13,7 +13,7 @@ export default function () {
         // Example: Send binary protocol header
         // Format: [0x01, length, ...data]
         const message = "Binary data";
-        const messageBytes = new TextEncoder().encode(message);
+        const messageBytes = Array.from(message).map(c => c.charCodeAt(0));
 
         const packet = new Uint8Array(2 + messageBytes.length);
         packet[0] = 0x01; // Protocol version

--- a/tcp/socket_write.go
+++ b/tcp/socket_write.go
@@ -56,13 +56,13 @@ func (s *socket) writeAsync(data sobek.Value, opts *writeOptions) (*sobek.Promis
 }
 
 func (s *socket) writePrepare(input sobek.Value, opts *writeOptions) ([]byte, *writeOptions, error) {
-	data, err := stringOrArrayBuffer(input, s.vu.Runtime())
-	if err != nil {
-		return nil, nil, err
-	}
-
 	if opts == nil {
 		opts = &writeOptions{}
+	}
+
+	data, err := stringOrArrayBuffer(input, s.vu.Runtime())
+	if err != nil {
+		return nil, opts, err
 	}
 
 	return data, opts, nil

--- a/tcp/socket_write.go
+++ b/tcp/socket_write.go
@@ -137,6 +137,15 @@ func stringOrArrayBuffer(input sobek.Value, runtime *sobek.Runtime) ([]byte, err
 			return nil, err
 		}
 
+	case reflect.TypeFor[sobek.ArrayBuffer]():
+		var ab sobek.ArrayBuffer
+
+		if err := runtime.ExportTo(input, &ab); err != nil {
+			return nil, err
+		}
+
+		data = ab.Bytes()
+
 	default:
 		return nil, fmt.Errorf("%w: String or ArrayBuffer expected", errInvalidType)
 	}

